models/DeliveryExecutionFinderService.php

```diff
@@ -67,15 +60,21 @@ public function findDeliveryExecution(LtiLaunchData $data): DeliveryExecution
             throw new LtiInvalidLaunchDataException('Wrong result ID provided');
```

Comment:
`deliveryExecutionId` cannot be null as far as I see, only if `$data[self::DELIVERY_EXECUTION_ID]` or  `$data->getVariable(self::LTI_SOURCE_ID)` evaluates to null which seems unlikely for me. Maybe an `empty()` check would be more accurate here. Also the exception message is misleading, because it's not wrong, it does not exist at all.